docs: update README (#36)
autowarefoundation/autoware.core: README.md

# autoware.core

> An [Autoware](https://github.com/autowarefoundation/autoware) repository that contains a basic set of high-quality, stable ROS packages for autonomous driving.

Reviewer: ```suggestion
- An [Autoware](https://github.com/autowarefoundation/autoware) repository that contains a basic set of high-quality, stable ROS packages for autonomous driving.
```

> For a more detailed explanation about Autoware Core, refer to the [Autoware concepts documentation page](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/#the-core-module).
> 
> For researchers and developers who want to extend the functionality of Autoware Core with experimental, cutting-edge ROS packages, see [Autoware Universe](https://github.com/autowarefoundation/autoware.universe).
> 
> We're discussing the interfaces at the [Autoware Architecture WG](https://github.com/autowarefoundation/autoware/discussions?discussions_q=label%3Aarchitecture_wg) now, and then we'll start porting the code from Universe to Core.
> 
> Note that, the code will be ported from Universe to Core after the interface discussion at [Autoware Architecture WG](https://github.com/autowarefoundation/autoware/discussions?discussions_q=label%3Aarchitecture_wg) has settled down.

Reviewer: - Added bullet points and rearranged the sentences. Does this still make sense?
- Removed line 11, since it's saying the same as line 9 did.
- I assume that in the future, we will remove the line about porting code, once that porting begins? 🤔 

```suggestion
- Although this repository is currently empty, porting of code from Universe to Core will begin once the interfaces for Autoware Core/Universe have been finalized, as per ongoing [Autoware Architecture WG](https://github.com/autowarefoundation/autoware/discussions?discussions_q=label%3Aarchitecture_wg) discussions. 
- A more detailed explanation about Autoware Core can be found on the [Autoware concepts documentation page](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/#the-core-module).

- For researchers and developers who want to extend the functionality of Autoware Core with experimental, cutting-edge ROS packages, see [Autoware Universe](https://github.com/autowarefoundation/autoware.universe).
```

# autoware.core

> A core repository of [Autoware](https://github.com/autowarefoundation/autoware), high-quality, stable ROS packages for Autonomous Driving.

Reviewer: Although I feel that `core` is the right word to use, using it to describe a repository called `Autoware Core` feels a little odd. 😅 

This is a suggestion but I am not 100% confident about it, so feedback is welcome! I think that we should also include a note to explain that we are currently developing the bus ODD using Autoware Universe and will be "porting" code from Universe to Core once code cleanup tasks have been completed. Any thoughts on this? @shmpwk @kminoda @kenji-miyake. 

```suggestion
The flagship [Autoware](https://github.com/autowarefoundation/autoware) repository, containing high-quality, stable ROS packages for autonomous driving.
```

Reviewer 2: I agree that the term `core` feels a bit confusing in this context, given that there exists `autoware.core` in AWF. 
As I don't know the details of the "porting" schedule, let me leave the decision to @kenji-miyake.

Author: `core` is a bit confusing but `flagship` is also confusing and abstract.
I am looking for specific and more precise words, but I can't come up with good word: `unchangeable`, `stable`, `main`, `confirmed`...... hmm.

Reviewer 3: > we should also include a note to explain that we are currently developing the bus ODD using Autoware Universe and will be "porting" code from Universe to Core once code cleanup tasks have been completed. Any thoughts on this?

Agreed. But before that, we should explain the concepts and key points of Core/Universe.

They are explained [here](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/
), for example: 
- Has basic features, but not high functionality
- Maintained by the AWF
- More strict criteria than Universe

> As I don't know the details of the "porting" schedule,

I think there is no need to write a concrete timeline.
Just saying like "we're discussing the interfaces at the [Autoware Architecture WG](https://github.com/autowarefoundation/autoware/discussions?discussions_q=label%3Aarchitecture_wg) now, and after that we'll start porting code from Universe to Core" is enough.

Author: Thank you for the comments.
To sum up those review, 2 points should be written.
1. brief description for autoware.core
2. the porting plan 

How about this?

ex.1 
```
Autoware Core has basic functionality and contains high-quality, stable ROS packages for [Autoware](https://github.com/autowarefoundation/autoware).

Universe code is initially experimental. After it is stable, we'll start porting the code from the Universe to Core.
```

ex.2
```
Autoware Core has  
- basic features, but not high functionality.
- high-quality, stable ROS packages
- AWF maintainers
- more strict criteria than Universe

We're discussing the interfaces at the [Autoware Architecture WG](https://github.com/autowarefoundation/autoware/discussions?discussions_q=label%3Aarchitecture_wg) now, and then we'll start porting the code from Universe to Core.
```

Reviewer 2: Thanks! Overall, I agree with @shmpw 's idea. 

One minor comment: ex.1 and ex.2 both sound like the explanation of "autoware.core" and "autoware.universe". So how about something like this?

===

The flagship [Autoware](https://github.com/autowarefoundation/autoware) repository, containing high-quality, stable ROS packages for autonomous driving.

Note that, in the near future, the code here in Universe will be ported to [Core](https://github.com/autowarefoundation/autoware.core).
Core is maintained by the Autoware Foundation with more strict criteria than Universe, having basic features with minimum functionality.

AWF is discussing the interfaces at the [Autoware Architecture WG](https://github.com/autowarefoundation/autoware/discussions?discussions_q=label%3Aarchitecture_wg) now, 
and once the interfaces are determined, Universe will be ported to Core.

Reviewer: @kminoda The explanation looks good to me, but there seems to have been a mix-up. This PR is to update the Autoware Core README not Autoware Universe, so the explanation will need to be revised. 

Since Mitsudome-san's [explanation of Core and Universe](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/difference-from-ai-and-auto/#autoware-coreuniverse) is now merged into the documentation, we can include that in the description.

Reviewer 2: @LalithVipulananthan You are right, thank you for pointing that out.

Author: Thank you for the comment @kminoda @LalithVipulananthan .

There are already sufficient infomation about Core/Universe [here](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/#the-core-module) and there is also comparison of them [here](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/difference-from-ai-and-auto/#autoware-coreuniverse). 
So IMO, just mentioning Core is enough in this README?

```
The flagship [Autoware](https://github.com/autowarefoundation/autoware) repository, containing high-quality, stable ROS packages for autonomous driving. 
The more explanation can be found [here](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-concepts/#the-core-module).
```

Author: @LalithVipulananthan 
Sounds perfect!

> For researchers and developers who extend the functionality of Autoware Core, see [Autoware.Universe](https://github.com/autowarefoundation/autoware.universe).

Reviewer: ```suggestion
For researchers and developers who want to extend the functionality of Autoware Core with experimental, cutting-edge ROS packages, see [Autoware Universe](https://github.com/autowarefoundation/autoware.universe).
```
